app/models/veteran_onboarding.rb

```diff
@@ -17,16 +17,42 @@ class VeteranOnboarding < ApplicationRecord
   belongs_to :user_account, primary_key: :id, foreign_key: :user_account_uuid, inverse_of: :veteran_onboarding
   validates :user_account, uniqueness: true
 
+  attr_accessor :user
+
   # Determines whether the onboarding flow should be displayed for a veteran.
-  # Currently, this is based solely on the value of the `display_onboarding_flow` attribute.
-  # In the future, additional information (like MPI info) might be taken into account.
+  # Currently, we have two feature toggle checks:
+  # - veteran_onboarding_show_to_newly_onboarded examines settings for veteran_onboarding.onboarding_threshold_days,
+  #   (default 180 days) and examines the user_verification.verified_at attribute
+  # - veteran_onboarding_beta_flow just checks if the feature toggle is enabled for the user
   def show_onboarding_flow_on_login
-    display_onboarding_flow
+    if @user.user_account&.verified?
+      if Flipper.enabled?(:veteran_onboarding_show_to_newly_onboarded, @user)
+        days_since_verification = UserVerification.where(user_account_id: @user.user_account_uuid).map do |uv|
+          (Time.zone.today - uv.verified_at.to_date).to_i
+        end.max
```

Comment:
Now check and find the oldest `UserVerification` record associated with the user - in case they have logged in with different verified accounts.